### PR TITLE
`service` nits

### PR DESCRIPTION
**Pull request by dhrubabasu**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,11 +19,13 @@ require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/btcsuite/btcd v0.22.0-beta // indirect
 	github.com/btcsuite/btcutil v1.0.3-0.20201208143702-a53e38424cce // indirect
+	github.com/cenkalti/backoff v2.2.1+incompatible // indirect
 	github.com/cespare/xxhash/v2 v2.1.1 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/deckarep/golang-set v1.7.1 // indirect
 	github.com/decred/dcrd/dcrec/secp256k1/v3 v3.0.0 // indirect
 	github.com/edsrzf/mmap-go v1.0.0 // indirect
+	github.com/fatih/color v1.10.0 // indirect
```
Comment (learyce) [inline go.mod:28]: Are we actually using these new packages?

File: service/service_account.go [hunk 2]
```diff
@@ -42,30 +43,30 @@ func (s AccountService) AccountBalance(
 		return nil, wrapError(errInvalidInput, "account identifier is not provided")
 	}
 
-	header, err := blockHeaderFromInput(ctx, s.client, req.BlockIdentifier)
-	if err != nil {
-		return nil, wrapError(errInternalError, err)
+	header, terr := blockHeaderFromInput(ctx, s.client, req.BlockIdentifier)
```
Comment (learyce) [inline service/service_account.go:46]: Why make these error vars less descriptive?

Comment (dhrubabasu) [reply service/service_account.go:46]: It's a stylistic choice, we don't do this anywhere else

File: service/service_account.go [hunk 3]
```diff
@@ -106,16 +106,14 @@ func (s AccountService) AccountBalance(
 		balances = append(balances, amount)
 	}
 
-	resp := &types.AccountBalanceResponse{
+	return &types.AccountBalanceResponse{
```
Comment (learyce) [inline service/service_account.go:109]: It's better to declare, then return when it's a complex object

File: service/service_account.go [hunk 4]
```diff
@@ -76,12 +77,11 @@ func (s AccountService) AccountBalance(
 	for _, currency := range req.Currencies {
 		value, ok := currency.Metadata[mapper.ContractAddressMetadata]
 		if !ok {
-			if types.Hash(currency) == types.Hash(mapper.AvaxCurrency) {
+			if utils.Equal(currency, mapper.AvaxCurrency) {
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply service/service_account.go:80]: https://github.com/coinbase/rosetta-sdk-go/blob/master/utils/utils.go#L111-L115

File: service/service_account.go [hunk 5]
```diff
@@ -91,7 +91,7 @@ func (s AccountService) AccountBalance(
 
 		data, err := hexutil.Decode(BalanceOfMethodPrefix + identifierAddress)
 		if err != nil {
-			return nil, wrapError(errCallInvalidParams, fmt.Errorf("failed to decode contractAddress in metadata field"))
+			return nil, wrapError(errCallInvalidParams, fmt.Errorf("marshalling balanceOf call msg data failed with %w", err))
```
Comment (patrick-ogrady) [inline service/service_account.go:94]: should be `%w: xxxxx`

File: service/service_account.go [hunk 6]
```diff
@@ -76,12 +77,11 @@ func (s AccountService) AccountBalance(
...
 		value, ok := currency.Metadata[mapper.ContractAddressMetadata]
 		if !ok {
-			if types.Hash(currency) == types.Hash(mapper.AvaxCurrency) {
+			if utils.Equal(currency, mapper.AvaxCurrency) {
 				balances = append(balances, mapper.AvaxAmount(avaxBalance))
 				continue
 			}
-			return nil, wrapError(errCallInvalidParams,
-				fmt.Errorf("currencies outside of avax must have contractAddress in metadata field"))
+			return nil, wrapError(errCallInvalidParams, fmt.Errorf("currencies outside of avax must have contractAddress in metadata field"))
```
Comment (patrick-ogrady) [inline service/service_account.go:84]: don't use `fmt` if no dynamic error used

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**